src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -95,6 +95,61 @@ const defaultProps = {
 };
 
 class SidebarLinks extends React.Component {
+    shouldComponentUpdate(nextProps) {
+        // We don't need to limit draft comment flashing for small screen widths as LHN is not visible.
+        if (nextProps.isSmallScreenWidth) {
+            return true;
+        }
+
+        const didActiveReportChange = this.props.currentlyViewedReportID !== nextProps.currentlyViewedReportID;
+        if (didActiveReportChange) {
+            return true;
+        }
+
+        const previousDraftComments = this.props.draftComments;
+        const nextDraftComments = nextProps.draftComments;
```

Comment:
```suggestion
        }
        
        // We only want to change the order of reports in LHN on larger screens,
        // when user changes report to avoid distractions while user is still composing a message.
        const previousDraftComments = this.props.draftComments;
        const nextDraftComments = nextProps.draftComments;
```

Reply:
Added a top-level comment to give general overview of what we want, and relevant comments near specific conditionals.